frontend/src/BenchmarkTest.res

```diff
@@ -9,13 +9,72 @@ type testMetrics = {
 
 @module("../icons/branch.svg") external branchIcon: string = "default"
 
-let decodeMetricValue = (json): LineGraph.DataRow.value => {
+let isSize = (x) => Js.Re.exec_(%re("/(gb|mb|kb|bytes)\w*/i"), x)
+  ->Belt.Option.isSome
+
+let formatSize = (value, units) => {
+  let str = Js.Float.toExponential(value)
+
+  // separate exponent into integral and exponent
+  // For example if the number is 314.00 then the exponential looks like
+  // 314.00 -> 3.14e+2
+  // where exp_ == 2
+  // and integral == 3.14
+  // the unit only rolls over if ceiling(n / 3) > 0
+  // so 3.14e+2 would clearly not have any change in units 
+  // as ceiling(2 / 3) == 0
+  // multiply integral (3.14) with 10^(mod(exp_, 3))
+  // in order to maintain correctness with the unit
+
+  let exp_ = Js.String.split("e", str) -> Belt.Array.getExn(1)
+  let sign = Js.String.get(exp_, 0)
+  let exp_ = (Js.String.get(exp_, 1)
+      ->Belt.Int.fromString
+      ->Belt.Option.getExn)
+  let exp = exp_ / 3
+  let integral = Js.String.split("e", str) 
+    -> Belt.Array.getExn(0)
+    -> Belt.Float.fromString
+    -> Belt.Option.map((x) => x *. (Js.Math.pow_float(~base=10.0, ~exp=mod(exp_,3)->Belt.Int.toFloat)))
+    -> Belt.Option.getExn
+  Js.log(integral)
+  let newValue = integral
+
+  let unitArr = ["bytes", "kb", "mb", "gb", "tb", "pb", "eb", "zb", "yb"]
+  let startIndex = Js.Re.exec_(%re("/(gb|mb|kb|bytes)\w*/i"), units)
+    ->Belt.Option.getExn
+    ->Js.Re.index
+  let endIndex = startIndex + 2
```

Comment:
This would be incorrect if the matched substring is `bytes`

Reply:
Thanks for the catch, I've replace the issue to now replace the string with the size of the matched string